src/game/behaviors/blue_fish.inc.c

```diff
@@ -1,76 +1,131 @@
+            }
+            
+            // Set pitch velocity
+            if (o->oTimer < (o->oBlueFishRandomTime + 60) / 2) {
                 o->oFaceAnglePitch += o->oAngleVelPitch;
-            else
+            } else {
                 o->oFaceAnglePitch -= o->oAngleVelPitch;
+            }
+            
+            // Calculate new Y velocity
             o->oVelY = -sins(o->oFaceAnglePitch) * o->oForwardVel;
             break;
-        case 1:
+        
+        // Animates and adjusts fish yaw angle.
+        case BLUE_FISH_ACT_TURN:
             func_8029ED98(0, 2.0f);
-            o->oMoveAngleYaw = (s32)(o->oBlueFishUnk100 + o->oMoveAngleYaw);
-            if (o->oTimer == 15)
+            o->oMoveAngleYaw = (s32)(o->oBlueFishRandomAngle + o->oMoveAngleYaw);
+            if (o->oTimer == 15) {
                 o->oAction++;
+            }
             break;
-        case 2:
+            
+        // Animates and adjusts pitch velocity.
+        case BLUE_FISH_ACT_ASCEND:
             func_8029ED98(0, 1.0f);
-            if (o->oTimer >= o->oBlueFishUnkF8 + 60)
+            
+            // Progresses oAction to BLUE_FISH_ACT_TURN_BACK after elapsed time. 
+            if (o->oTimer >= o->oBlueFishRandomTime + 60) {
                 o->oAction++;
-            if (o->oTimer < (o->oBlueFishUnkF8 + 60) / 2)
+            }
+            
+            // Adjusts pitch angle. Direction relies on time not passed.
+            if (o->oTimer < (o->oBlueFishRandomTime + 60) / 2) {
                 o->oFaceAnglePitch -= o->oAngleVelPitch;
-            else
+            } else {
                 o->oFaceAnglePitch += o->oAngleVelPitch;
+            }
             break;
-        case 3:
+            
+        // Animates and turns fish around
+        case BLUE_FISH_ACT_TURN_BACK:
             func_8029ED98(0, 2.0f);
-            o->oMoveAngleYaw = (s32)(o->oBlueFishUnk100 + o->oMoveAngleYaw);
-            if (o->oTimer == 15)
-                o->oAction = 0;
+            o->oMoveAngleYaw = (s32)(o->oBlueFishRandomAngle + o->oMoveAngleYaw);
+            
+            // Sets oAction to the BLUE_FISH_ACT_DIVE if the timer equals fifteen.
+            if (o->oTimer == 15) {
+                o->oAction = BLUE_FISH_ACT_DIVE;
+            }
             break;
     }
+    
+    // Calculates Y velocity and calls physics engine.
     o->oVelY = -sins(o->oFaceAnglePitch) * o->oForwardVel;
     obj_move_using_fvel_and_gravity();
-    if (o->parentObj->oAction == 2)
+    
+    // Deletes object if the parent has oAction set to two.
+    if (o->parentObj->oAction == BLUE_FISH_ACT_ASCEND) {
         mark_object_for_deletion(o);
+    }
 }
 
-void bhv_tank_fish_group_loop(void) {
+/**
+ * Spawns fifteen fish if mario resides in room fifteen or seven.
+ * It moves at random within 200.0f
+ */
+void bhv_blue_fish_spawn_loop(void) {
     struct Object *fish;
     s32 i;
     switch (o->oAction) {
-        case 0:
+        case BLUE_FISH_ACT_DIVE:
```

Comment:
this spawn loop is used by a different object, so it needs to use different names. (I'd change the function back to fish_tank or something as well.)

Reply:
bhv_tank isn't consistent with the other method names of blue_fish_ and tank doesn't really describe what the function is.
Unless the clarification of tank helps to differentiate with other fish methods?

Reply:
It's fine if it's not consistent with blue_fish_, because it's a different object. Sorry if I was unclear, this is the main loop for `bhvTankFishGroup`, which is what spawns the fish in the JRB room. To be consistent with the rest of the code base, this function should be named `bhv_tank_fish_group_loop`. (or rename bhvTankFishGroup to something better like `bhvBlueFishTank`/`Spawn`)

But because this is the TankFishGroup's (or whatever you want to rename it)'s main loop, `BLUE_FISH_ACT_DIVE` and etc. shouldn't be reused in this function.